src/selection.ts

```diff
@@ -73,8 +74,15 @@ type getFragmentSelection<
   Type extends ObjectLikeType,
   Introspection extends IntrospectionLikeType,
   Fragments extends { [name: string]: any },
+  UnionAccumulator,
```

Comment:
I don't really know btw what this generic is trying to achieve. It's possible you had a performance concern here, or something else, but the name and placing of this doesn't make it immediately obvious ❤️